app/streaming/input/abstouch.cpp

```diff
@@ -115,7 +137,23 @@ void SdlInputHandler::handleAbsoluteFingerEvent(SDL_TouchFingerEvent* event)
     // Try to send it as a native pen/touch event, otherwise fall back to our touch emulation
     if (LiGetHostFeatureFlags() & LI_FF_PEN_TOUCH_EVENTS) {
         bool isPen = false;
+#if SDL_VERSION_ATLEAST(3, 0, 0)
+        int numTouchDevices = 0;
+        SDL_TouchID *touchId = SDL_GetTouchDevices(&numTouchDevices);
+        for (int i = 0; i < numTouchDevices; i++) {
+            if (event->touchId == touchId[i]) {
+                const char* touchName = SDL_GetTouchDeviceName(touchId[i]);
 
+                // SDL will report "pen" as the name of pen input devices on Windows.
```

Comment:
Actual pen support (using Windows Ink for now, WinTab support coming later) just landed in SDL3, so if you want _actual_ pens and not touches, you should look for SDL_EVENT_PEN_* events.

Pens are a mess; in most cases we should be able to tell you all sorts of stuff about the hardware that's available, but most operating systems just sort of surprise you with the presence of a pen when it starts sending input events about it, and won't tell you anything _about_ the pen's abilities. As such, we built SDL's pen support to match that. Take a look at [the details](https://wiki.libsdl.org/SDL3/CategoryPen).